### ISSUE TITLE
Allow google OAuth credentials

### ISSUE THREAD

**Issue by etnbrd** [cozy/cozy-stack#784]
I am in the cozy meetup to create a Konnector, and decided to go with google calendar.
It seems the authentification will be an issue to deploy and publish the konnector.
Would it be possible to integrate this authentification directly into the stack itself ? (Managing the secret token, and the per-client authorisation)
I heard that a mechanism was developed for MAIF and Orange, would it be possible to duplicate them for google services ?

**Comment by clochix** [comment]
Hello @etnbrd,

Following the meeting, we discussed of the need to allow connectors to use OAuth to connect to remote services.
This is on our roadmap, but it's quite difficult to implement, so it probably won't be available before a few months.

/cc @doubleface

**Comment by etnbrd** [comment]
How can we help ?

**Comment by aenario** [comment]
Hello @etnbrd, sorry for the delay.

We made some changes which allows support for strict oauth like google does, ony setting up one google oauth client for the whole domain (and not one / cozy)

## Setting up the client with google
Go to : https://console.developers.google.com/apis/credentials
Create Project
credentials -> Create credentials -> Oauth Client ID

## Setting up the client with cozy
```
http PUT localhost:5984/secrets%2Fio-cozy-accounts_types/google
client_id=$CLIENT_ID
client_secret=$CLIENT_SECRET
grant_mode=authorization_code
redirect_uri="https://oauthcallback.cozy.tools:8080/accounts/google/redirect" 
token_endpoint="https://www.googleapis.com/oauth2/v4/token"
auth_endpoint="https://accounts.google.com/o/oauth2/v2/auth"
```

## Begin oauth flow for some scopes
```
somecozy.cozy.tools:8080/accounts/google/start?scope=profile+email
```

If you have a konnector that works with this account configured, we can ask our devops to setup this account_types with Cozy Cloud Company secrets in our infra.

**Comment by nono** [comment]
Fixed by #888
